src/network/taraxa_capability.cpp

```diff
@@ -257,27 +257,28 @@ void TaraxaCapability::interpretCapabilityPacketImpl(NodeID const &_nodeID, unsi
       break;
     }
     case StatusPacket: {
-      bool initial_status = _r.itemCount() == 9;
+      bool initial_status = _r.itemCount() == INITIAIL_STATUS_PACKET_ITEM_COUNT;
 
       if (initial_status) {
         auto it = _r.begin();
         auto const network_id = (*it++).toInt<uint64_t>();
-        auto peer_dag_level = (*it++).toPositiveInt64();
+        auto const peer_dag_level = (*it++).toPositiveInt64();
         auto const genesis_hash = blk_hash_t(*it++);
-        auto peer_pbft_chain_size = (*it++).toPositiveInt64();
-        auto peer_syncing = (*it++).toInt();
-        auto peer_pbft_round = (*it++).toPositiveInt64();
-        auto peer_pbft_previous_round_next_votes_size = (*it++).toInt<unsigned>();
-        auto node_major_version = (*it++).toInt();
-        auto node_minor_version = (*it++).toInt();
+        auto const peer_pbft_chain_size = (*it++).toPositiveInt64();
+        auto const peer_syncing = (*it++).toInt();
+        auto const peer_pbft_round = (*it++).toPositiveInt64();
+        auto const peer_pbft_previous_round_next_votes_size = (*it++).toInt<unsigned>();
+        auto const node_major_version = (*it++).toInt<unsigned>();
+        auto const node_minor_version = (*it++).toInt<unsigned>();
+        auto const node_patch_version = (*it++).toInt<unsigned>();
 
         // We need logic when some different node versions might still be compatible
-        if (node_major_version != FullNode::c_node_major_version ||
-            node_minor_version != FullNode::c_node_minor_version) {
-          LOG(log_er_) << "Incorrect node version: " << getFormattedVersion(node_major_version, node_minor_version)
-                       << ", our node major version"
-                       << getFormattedVersion(FullNode::c_node_major_version, FullNode::c_node_minor_version)
-                       << ", host " << _nodeID.abridged() << " will be disconnected";
+        if (node_major_version != TARAXA_MAJOR_VERSION || node_minor_version != TARAXA_MINOR_VERSION ||
+            node_patch_version != TARAXA_PATCH_VERSION) {
```

Comment:
I think we should NOT enforce the patch version having to be equal, and anytime we want to FORCE nodes to upgrade we change the minor version AND/OR the `TARAXA_NET_VERSION`.   By the time we hit mainnet we will remove this check entirely because interoperability will be defined entirely by protocol version and block number in mainnet.

Reply:
fixed